src/lumen/activity/field.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include "lumen/activity/activity.hpp"
+
+#include <string>
+
+namespace lumen::activity {
+
+class Activity;
+
+/// Base class from which displayable types derive.
+template <typename T>
+class Field {
+public:
+    /** Field constructor.
+     *
+     * \param parent Pointer to the activity object that should be updated when
+     * a state change occurs.
+     *
+     * \param value Initial value of the field.
+     */
+    Field(Activity* parent, T value) : parent_{parent}, value_{value} {}
+    virtual ~Field() = default;
+
+    /// Get the current value of the field.
+    [[nodiscard]] T get_value() const
```

Comment:
- [ ] This should be defined in `field.cpp`

Reply:
Templates need to be defined in the header file.